src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/DuplicateTypeNameAnalyzer.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Azure.ClientSdk.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class DuplicateTypeNameAnalyzer : SymbolAnalyzerBase
+    {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Descriptors.AZC0034);
+        public override SymbolKind[] SymbolKinds { get; } = { SymbolKind.NamedType };
+
+        // Common .NET platform type names that Azure SDK types should avoid conflicting with
+        private static readonly HashSet<string> PlatformTypeNames = new HashSet<string>
```

Comment:
The list of types here is nowhere to be complete. Any ideas how to address that? Maybe create a large file with all single word and two word type names from the whole Azure SDK, sort the names in the file, and then binary search them?